tests/tensor_search/test_on_start_script.py

```diff
@@ -47,7 +47,111 @@ def run():
+        ]
+        for mock_environ, expected in environ_expected_models:
+            mock_vectorise = mock.MagicMock()
+            @mock.patch("os.environ", mock_environ)
+            @mock.patch("marqo.tensor_search.on_start_script.vectorise", mock_vectorise)
+            def run():
+                model_caching_script = on_start_script.ModelsForCacheing()
+                model_caching_script.run()
+                loaded_models = {
+                    (
+                        kwargs["model_name"],
+                        kwargs["model_properties"]["name"],
+                        kwargs["model_properties"]["dimensions"],
+                        kwargs["model_properties"]["type"],
+                        kwargs["model_properties"]["url"]
+                    )
+                    for args, kwargs in mock_vectorise.call_args_list
+                }
+                assert loaded_models == set(expected)
+                return True
+            assert run()
+    
+    def test_preload_url_missing_model(self):
+        open_clip_model_object = {
+            "model_properties": {
+                "name": "ViT-B-32-quickgelu",
+                "dimensions": 512,
+                "type": "open_clip",
+                "url": "https://github.com/mlfoundations/open_clip/releases/download/v0.2-weights/vit_b_32-quickgelu-laion400m_avg-8a00ab3c.pt"
+            }
+        }
+        mock_vectorise = mock.MagicMock()
+        @mock.patch("marqo.tensor_search.on_start_script.vectorise", mock_vectorise)
+        @mock.patch("os.environ", {enums.EnvVars.MARQO_MODELS_TO_PRELOAD: [open_clip_model_object]})
+        def run():
+            try:
+                model_caching_script = on_start_script.ModelsForCacheing()
+                # There should be a KeyError -> EnvVarError when attempting to call vectorise
+                model_caching_script.run()
+                raise AssertionError
+            except errors.EnvVarError as e:
+                return True
+        assert run()
+    
+    def test_preload_url_missing_model_properties(self):
+        open_clip_model_object = {
+            "model": "random-open-clip-1"
+        }
+        mock_vectorise = mock.MagicMock()
+        @mock.patch("marqo.tensor_search.on_start_script.vectorise", mock_vectorise)
+        @mock.patch("os.environ", {enums.EnvVars.MARQO_MODELS_TO_PRELOAD: [open_clip_model_object]})
+        def run():
+            try:
+                model_caching_script = on_start_script.ModelsForCacheing()
+                # There should be a KeyError -> EnvVarError when attempting to call vectorise
+                model_caching_script.run()
+                raise AssertionError
+            except errors.EnvVarError as e:
+                return True
+        assert run()
+    
+    # TODO: test bad/no names/URLS in end-to-end tests, as this logic is done in vectorise call
 
```

Comment:
can we have one more test to ensure the prewarmed model will not need to be download/load again?

Reply:
this we can save for the API end-to-end test